cmd/podman/ps.go

```diff
@@ -506,41 +579,51 @@ func getTemplateOutput(containers []*libpod.Container, opts batchcontainer.PsOpt
 		}
 		psOutput = append(psOutput, params)
 	}
-	// Sort the ps entries by created time
-	sort.Sort(psSorted(psOutput))
 
 	return psOutput, nil
 }
 
 // getJSONOutput returns the container info in its raw form
-func getJSONOutput(containers []*libpod.Container, opts batchcontainer.PsOptions) ([]psJSONParams, error) {
+func getAndSortParams(containers []*libpod.Container, opts batchcontainer.PsOptions) (psSorted, error) {
 	var (
-		psOutput []psJSONParams
+		psOutput psSorted
 		ns       *batchcontainer.Namespace
 	)
 	for _, ctr := range containers {
 		batchInfo, err := batchcontainer.BatchContainerOp(ctr, opts)
 		if err != nil {
 			return nil, err
 		}
+
+		var createArtifact cc.CreateConfig
+		artifact, err := ctr.GetArtifact("create-config")
+		if err == nil {
+			if err := json.Unmarshal(artifact, &createArtifact); err != nil {
+				return nil, err
+			}
+		} else {
+			logrus.Errorf("couldn't get some ps information, error getting artifact %q: %v", ctr.ID(), err)
+		}
+		mounts := createArtifact.Volumes
 		if opts.Namespace {
 			ns = batchcontainer.GetNamespaces(batchInfo.Pid)
 		}
-		params := psJSONParams{
-			ID:               ctr.ID(),
-			Image:            batchInfo.ConConfig.RootfsImageName,
-			ImageID:          batchInfo.ConConfig.RootfsImageID,
-			Command:          batchInfo.ConConfig.Spec.Process.Args,
-			CreatedAt:        batchInfo.ConConfig.CreatedTime,
-			Status:           batchInfo.ConState.String(),
-			Ports:            batchInfo.ConConfig.PortMappings,
-			RootFsSize:       batchInfo.RootFsSize,
-			RWSize:           batchInfo.RwSize,
-			Names:            batchInfo.ConConfig.Name,
-			Labels:           batchInfo.ConConfig.Labels,
-			Mounts:           batchInfo.ConConfig.Spec.Mounts,
-			ContainerRunning: batchInfo.ConState == libpod.ContainerStateRunning,
-			Namespaces:       ns,
+		params := psParams{
+			ID:         ctr.ID(),
+			Image:      batchInfo.ConConfig.RootfsImageName,
+			ImageID:    batchInfo.ConConfig.RootfsImageID,
+			Command:    batchInfo.ConConfig.Spec.Process.Args,
+			CreatedAt:  batchInfo.ConConfig.CreatedTime,
+			ExitCode:   batchInfo.ExitCode,
+			Status:     batchInfo.ConState,
+			PID:        batchInfo.Pid,
+			Ports:      batchInfo.ConConfig.PortMappings,
+			RootFsSize: batchInfo.RootFsSize,
+			RWSize:     batchInfo.RwSize,
+			Names:      batchInfo.ConConfig.Name,
+			Labels:     batchInfo.ConConfig.Labels,
+			Mounts:     mounts,
```

Comment:
You know, we can probably use `batchInfo.ConConfig.UserVolumes` instead for this, and drop the createArtifact entirely. Speed things up a fair bit because we don't need to read the artifact anymore.